### PR TITLE
fix: auto-dismiss MP bottom sheet after Buy Crypto click

### PR DESCRIPTION
**Pull request by just-toby**
## Description

https://uniswaplabs.atlassian.net/browse/WEB-3155?atlOrigin=eyJpIjoiN2I2NzkxMTcyY2NhNDgzNDkxMWQzZDBmMjMwM2RjNWMiLCJwIjoiaiJ9

https://uniswapteam.slack.com/archives/C047U65H422/p1680546838383989

#### Manual

manually verified that the MP drawer/bottomsheet closes when you click buy crypto. and the same check in an e2e test

#### Automated
- [ ] Unit test
- [x] Integration/E2E test

### CODE REVIEW

**Comment by tinaszheng** [comment]
image
this has odd behavior in NYC - since it's not available in my region, the wallet drawer just closes when i click it and nothing happens. there should probably be a check first before dismissing

**Comment by tinaszheng** [comment]
@grabbou @zzmp since y'all are outside of NYC can you take a look too! 

the blocked flow looks good to me ✅

File: cypress/e2e/wallet-dropdown.test.ts [hunk 1]
```diff
@@ -77,4 +77,12 @@ describe('Wallet Dropdown', () => {
     cy.get(getTestSelector('theme-auto')).click()
     cy.get(getTestSelector('wallet-header')).should('have.css', 'color', 'rgb(119, 128, 160)')
   })
+
+  it('should dismiss the wallet bottom sheet when clicking buy crypto', () => {
+    visit(false)
+    cy.viewport('iphone-6')
```
Comment (JFrankfurt) [inline cypress/e2e/wallet-dropdown.test.ts:83]: didn't know we could do this! v cool

**Review by JFrankfurt — APPROVED** [review]
lgtm

**Review by JFrankfurt — APPROVED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**